lib/internal/test_runner/mock/loader.js

```diff
@@ -118,10 +118,17 @@ async function load(url, context, nextLoad) {
   // Treat builtins as commonjs because customization hooks do not allow a
   // core module to be replaced.
   // Also collapse 'commonjs-sync' and 'require-commonjs' to 'commonjs'.
-  const format = (
-    original.format === 'builtin' ||
-    original.format === 'commonjs-sync' ||
-    original.format === 'require-commonjs') ? 'commonjs' : original.format;
+  let format = original.format;
+  switch (original.format) {
+    case 'builtin': // Deliberate fallthrough
+    case 'commonjs-sync': // Deliberate fallthrough
+    case 'require-commonjs':
+      format = 'commonjs';
+      break;
+    case 'json':
+      format = 'module';
```

Comment:
This is the main change. Creating a nested ternary seemed messy, so I changed this to a switch.